### PR TITLE
Add hamburger button

### PR DESCRIPTION
**Pull request by senguyen1011**
Added a hamburger icon button. Animations are included in CSS, but require JS to toggle between hamburger (default) state and active state. Active states include a cross (X), minus sign, left arrow, and right arrow.

Issue: #765

### CODE REVIEW

File: src/components/icon/_hamburger.less [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    background: currentColor;
+    transform: translate(-50%, -50%);
+    transition: transform 0.2s ease;
+  }
+}
+
+.hamburger-btn:hover {
```
Comment (shahednasser) [inline src/components/icon/_hamburger.less:28]: nest inside `.hamburger-btn` in the beginning.

**Review by shahednasser — CHANGES_REQUESTED** [review]
Thank you for your work! I'm not sure why you have many is-active states. I think you just need the x one and the transition should just be by adding `.is-active` (without adding `hamburger-x-btn`). Also the transition between hamburger and x doesn't look so clean so please work on it.

File: src/components/icon/_hamburger.less (hunk 1 above)
Comment (senguyen1011) [reply src/components/icon/_hamburger.less:28]: `.hamburger-btn` is a parent class and other classes like `.hamburger-x-btn`, `.hamburger-min-btn`, `.hamburger-larr-btn` determine what icon the button will transform into when toggled. Also, could you tell me what issue you're finding with the transition? When I test it locally, it seems fine to me (there are 2 lines, :before and :after, hidden in the middle bar of the hamburger icon. They both turn 45 degrees to form the 'X' while the top and bottom bars fade away)
hamburger

Comment (shahednasser) [reply src/components/icon/_hamburger.less:28]: I understand but I think the most used is hamburger to x so let's keep it simple by just adding those two states. By just adding `.active` to `.hamburger-btn` it should change to the x. As for the transition I think it looks good so forget my comment about it.

**Comment by senguyen1011** [comment]
I've removed the variant hamburger icons, but my checks aren't passing and I'm unsure of how to fix that.

**Comment by shahednasser** [comment]
Great work thank you!

@all-contributors please add @senguyen1011 for code and idea

**Pull request merged**